src/physics/grid/NonuniformGrid.i.hh

```diff
@@ -0,0 +1,67 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file NonuniformGrid.i.hh
+//---------------------------------------------------------------------------//
+#include "base/Algorithms.hh"
+#include "base/Assert.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with data.
+ */
+template<class T>
+CELER_FUNCTION
+NonuniformGrid<T>::NonuniformGrid(const ItemRange<value_type>& values,
+                                  const Values&                data)
+    : data_(data[values])
+{
+    CELER_EXPECT(data_.size() >= 2);
+    CELER_EXPECT(data_.front() <= data_.back()); // Approximation for "sorted"
```

Comment:
Do we assume that the check that the data is sorted has already been done (like when the value grid is constructed)?

Reply:
I added the `ValueGridRangeBuilder` which has an extra 
```c++
    CELER_EXPECT(is_monotonic_increasing(this->value()));
```
so as long as we're using that to construct the range data we'll be OK.